util/snprintf.c

```diff
@@ -26,12 +27,12 @@ int vsnprintf(char* out, size_t n, const char* s, va_list vl)
               longarg = true;
           break;
         case 'p':
-          longarg = true;
+          uintptrarg = true;
           if (++pos < n) out[pos-1] = '0';
           if (++pos < n) out[pos-1] = 'x';
         case 'x':
         {
-          long num = longarg ? va_arg(vl, long) : va_arg(vl, int);
+          long num = uintptrarg ? va_arg(vl, uintptr_t) : (longarg ? va_arg(vl, long) : va_arg(vl, int));
           for(int i = 2*(longarg ? sizeof(long) : sizeof(int))-1; i >= 0; i--) {
```

Comment:
This will lose the high half of the address currently.

Reply:
```suggestion
          for(int i = 2*((longarg || uintptrarg) ? sizeof(long) : sizeof(int))-1; i >= 0; i--) {
```